src/main/java/executor/service/service/impl/parallel/TaskWorker.java

```diff
@@ -0,0 +1,44 @@
+package executor.service.service.impl.parallel;
+
+import executor.service.service.Item;
+import executor.service.service.ItemHandler;
+import executor.service.service.Listener;
+
+import java.util.concurrent.Callable;
+import java.util.function.Consumer;
+
+/**
+ * The {@code TaskWorker} class is a generic implementation
+ * of the {@link Callable} interface,
+ * responsible for creating task workers
+ * that produce items and add them to a specified queue.
+ * <p>
+ *
+ * @author Oleksandr Tuleninov
+ * @version 01
+ * @see executor.service.service.Listener
+ * @see ItemQueue
+ * @see Consumer
+ */
+public class TaskWorker implements Callable<ItemQueue> {
+
+    private final Listener listener;
+
+    public TaskWorker(Listener listener) {
+        this.listener = listener;
+    }
+
+    @Override
+    public ItemQueue call() {
+        ItemQueue queue = new ItemQueue();
+        Consumer<Item> itemHandlerConsumer = queue::putItem;
+
+        listener.execute(createHandler(itemHandlerConsumer));
+
+        return queue;
+    }
+
+    private ItemHandler<Item> createHandler(Consumer<Item> consumer) {
+        return consumer::accept;
+    }
```

Comment:
Are you sure  each time when method calls to create new queue?

Reply:
1. Why are you writing this, in the program code this happens only once, look at the code carefully.
2. Additionally, I inform you that the code has been refactored and this section of code is not in the program.

Reply:
Can you add logging using sout how many times it happened? On my PC branch team. Doesn't run =)